src/Simplify.cpp

```diff
@@ -1444,23 +1470,48 @@ class Simplify : public IRMutator {
             mod_rem = modulus_remainder(ramp_a->base, alignment_info);
         }
 
-        if (is_zero(a) && !is_zero(b)) {
+        if (is_indeterminate_expression(call_a)) { 
             expr = a;
+        } else if (is_indeterminate_expression(call_b)) {
+            expr = b;
+        } else if (is_zero(b)) {
```

Comment:
I could agree that we shouldn't generate indeterminate expressions for floats, but the question of how to handle nonfinite values for floats is one worth debating. IIUC, -ffast-math in GCC implies also implies -ffinite-math-only, so we'd be within our rights to just throw our hands in the air and say "whatevs"... that said, it seems like detecting obvious gotchas at constant-folding time is low-hanging fruit (e.g. "Inf - Inf" shouldn't simplify to zero).

How about this: 
-- I rework this PR to only address the integer case (div-by-zero -> indeterminate)
-- I open a new issue documenting the fact that Simplify could be upgraded to handle non-finite float values better, so that it's at least discoverable, and a possible future work item

?